backends/core/workers.py

```diff
@@ -949,3 +950,38 @@ def _execute(self):
     job = client.run_async_query(job_name, self._params['query'])
     job.use_legacy_sql = False
     self._begin_and_wait(job)
+
+class AutoMLPredictor(BQWorker):
+  """Worker to run BQML SQL queries in BigQuery."""
+
+  PARAMS = [
+      ('bq_project_id', 'string', False, '', 'BQ Project ID'),
+      ('compute_region', 'string', False, '', 'Compute Region'),
+      ('model_id', 'string', False, '', 'AutoML Model ID'),
+      ('bq_dataset_id', 'string', True, '', 'BQ Dataset ID'),
+      ('bq_table_id', 'string', True, '', 'BQ Table ID'),
+      ('destination_uri', 'string', True, '',
+       'Destination CSV file path (e.g. gs://bucket/)'),
+  ]
+
+  def _execute(self):
+    self._bq_setup()
+    automl_client = automl.AutoMlClient()
+    # Get the full path of the model.
+    model_full_id = automl_client.model_path(
+        self._params['bq_project_id'], self._params['compute_region'], self._params['model_id']
+    )
+    # Create client for prediction service.
+    prediction_client = automl.PredictionServiceClient()
+    input_config = {"bigquery_source": {"input_uri": self._table}}
+    output_config = {"gcs_destination": {"output_uri_prefix": self._params['destination_uri']}}
+    # Query model
+    response = prediction_client.batch_predict(
+        model_full_id, input_config, output_config)
+    print("Making batch prediction... ")
+    try:
+        result = response.result()
+    except:
+        # Hides Any to BatchPredictResult error.
```

Comment:
Instead of silently failing, you can use `self. log_error(...)` or even better `raise WorkerException('Failed to get results with...')` which will trigger an automatic retry.